### PR TITLE
Handle Config Plugin Events

### PR DESCRIPTION
**Pull request by trusz**

### CODE REVIEW

File: packages/openscd/src/open-scd.ts [hunk 1]
```diff
@@ -278,15 +259,51 @@ export class OpenSCD extends LitElement {
...
+      this.changePlugin(config)
+
+    }else if( isRemoveEvent ){
+      this.removePlugin(name, kind)
+
+    }else if(isAddEvent && config){
+      this.addPlugin(config)
+
+    }else{
+      console.error("Invalid plugin configuration event", {name, kind, config})
```
Comment (clepski) [inline packages/openscd/src/open-scd.ts:297]: For logging we should use the log events, so the user can the error message

File: packages/openscd/src/open-scd.ts [hunk 2]
```diff
@@ -341,6 +360,51 @@ export class OpenSCD extends LitElement {
...
+    this.storePlugins(newPlugins);
+  }
+
+  private changePlugin(plugin: Plugin) {
+    const storedPlugins = this.storedPlugins
+    const {name, kind} = plugin
+    const pluginIndex = this.findPluginIndex(name, kind)
+
+    if(pluginIndex < 0) {
+      console.warn("Plugin not found, stopping.", {name, kind})
```
Comment (clepski) [inline packages/openscd/src/open-scd.ts:396]: Use log event

File: packages/openscd/src/open-scd.ts [hunk 3]
```diff
@@ -278,15 +259,51 @@ export class OpenSCD extends LitElement {
...
+    const isChangeEvent = hasPlugin && hasConfig
+    const isRemoveEvent = hasPlugin && !hasConfig
+    const isAddEvent = !hasPlugin && hasConfig
+
+    // the `&& config`is only because typescript
+    // cannot infer that `isChangeEvent` and `isAddEvent` implies `config !== null`
+    if(isChangeEvent && config){
+      this.changePlugin(config)
+
+    }else if( isRemoveEvent ){
```
Comment (clepski) [inline packages/openscd/src/open-scd.ts:290]: Code formatting: whitespaces, missing `;` etc

```
if (...) {
  doStuff;
}
```

File: packages/openscd/web-test-runner.config.mjs [hunk 4]
```diff
@@ -31,10 +37,10 @@ export default /** @type {import("@web/test-runner").TestRunnerConfig} */ ({
...
       files: 'test/integration/**/*.test.ts',
     },
-    {
-      name: 'single',
-      files: 'test/unit/foundation.test.ts',
-    },
+    // {
+    //   name: 'single',
+    //   files: 'test/unit/Plugging.test.ts',
+    // },
```
Comment (clepski) [inline packages/openscd/web-test-runner.config.mjs:43]: FYI you can start the tests like `npm run test:unit` and then select a single test file via the CLI with `F` and select the testfile

**Review by clepski — CHANGES_REQUESTED** [review]

File: packages/openscd/src/open-scd.ts (hunk 1 above)
Comment (trusz) [reply packages/openscd/src/open-scd.ts:297]: This is meant for developers, should the users see this?

Comment (clepski) [reply packages/openscd/src/open-scd.ts:297]: If this is something that should never happen unless something is wrong then I would throw an error

**Review by clepski — APPROVED** [review]

**Pull request merged**